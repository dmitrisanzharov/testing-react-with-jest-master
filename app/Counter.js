import React from 'react'

const Counter = () => {

    const [count, setCount] = React.useState(0)


    function makeIncrementer (amount) {
      setCount(count + amount)
    }

    function incrementFunction(){
      makeIncrementer(1);
    }

    function decrementFunction(){
      makeIncrementer(-1)
    }



  return (
    <div>
        <h1>counter</h1>
        <h2>the count is: {count}</h2>
        <button onClick={incrementFunction} id='button1'>Increase</button>
        <button onClick={decrementFunction} id='button2'>Decrease</button>
    </div>
  )
}

export default Counter