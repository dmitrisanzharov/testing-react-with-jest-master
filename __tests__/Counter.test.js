import React from 'react';
import Counter from '../app/Counter';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer'; 

describe.skip('All test', ()=> {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Counter />);
    })


    it('matches the snapshot', ()=> {
        const tree = renderer.create(<Counter />).toJSON();
        expect(tree).toMatchSnapshot();
    })



    it('starts with a count of 0', ()=> {
  
        const text = wrapper.find('h2').text();
        expect(text).toEqual('the count is: 0');
       
    });


    it('increments by 1', ()=> {

        const incrementBtn = wrapper.find('#button1');

        incrementBtn.simulate('click');
   
         const text = wrapper.find('h2').text();
      
        expect(text).toBe('the count is: 1');


    })

    it('decrements by 1', ()=> {

        const incrementBtn = wrapper.find('#button2');

        incrementBtn.simulate('click');
   
         const text = wrapper.find('h2').text();
      
        expect(text).toBe('the count is: -1');

        // console.log(wrapper.debug());
    })



    // end of a test
})