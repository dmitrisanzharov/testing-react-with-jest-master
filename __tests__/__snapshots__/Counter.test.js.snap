// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All test matches the snapshot 1`] = `
<div>
  <h1>
    counter
  </h1>
  <h2>
    the count is: 
    0
  </h2>
  <button
    id="button1"
    onClick={[Function]}
  >
    Increase
  </button>
  <button
    id="button2"
    onClick={[Function]}
  >
    Decrease
  </button>
</div>
`;
